GDJS/Runtime/AsyncTasksManager.ts

```diff
@@ -51,6 +96,66 @@ namespace gdjs {
     clearTasks() {
       this.tasksWithCallback.length = 0;
     }
+
+    getNetworkSyncData(
+      syncOptions: GetNetworkSyncDataOptions
+    ): AsyncTasksManagerNetworkSyncData {
+      const tasksData = this.tasksWithCallback.map(
+        ({ asyncTask, callbackId, longLivedObjectsList }) => {
+          return {
+            callbackId,
+            asyncTask: asyncTask.getNetworkSyncData(),
+            objectsList: longLivedObjectsList.getNetworkSyncData(syncOptions),
+          };
+        }
+      );
+
+      return {
+        tasks: tasksData,
+      };
+    }
+
+    updateFromNetworkSyncData(
+      syncData: AsyncTasksManagerNetworkSyncData,
+      idToCallbackMap: Map<
+        string,
+        (
+          runtimeScene: gdjs.RuntimeScene,
+          asyncObjectsList: gdjs.LongLivedObjectsList
+        ) => void
+      >,
+      runtimeScene: gdjs.RuntimeScene,
+      syncOptions: UpdateFromNetworkSyncDataOptions
+    ) {
+      this.clearTasks();
+
+      syncData.tasks.forEach(({ callbackId, asyncTask, objectsList }) => {
+        if (!asyncTask) return;
+
+        const callback = idToCallbackMap.get(callbackId);
+        if (callback) {
+          const longLivedObjectsList = new gdjs.LongLivedObjectsList();
+          longLivedObjectsList.updateFromNetworkSyncData(
+            objectsList,
+            runtimeScene,
+            syncOptions
+          );
+
+          if (asyncTask.type === 'group') {
+            const task = new TaskGroup();
+            task.updateFromNetworkSyncData(asyncTask);
+            this.addTask(task, callback, callbackId, longLivedObjectsList);
+          }
+          if (asyncTask.type === 'wait') {
```

Comment:
else if + else {
  unknownTaskTypes.add(task)
}

and a if (unknownTaskTypes.size > 0) console.warn("Some tasks could be not restored: ...").